### PR TITLE
Fix issues found by cppclean in libs, iroha-cli, irohad/ametsuchi, irohad/consensus

### PR DESCRIPTION
**Pull request by Warchant**
### Description of the Change

cppclean has found many issues (https://hackmd.io/IYTmBNwRmAGBaATANgOwICzIMaPgDgDMBWBfbYDAI1gGZbgrFag=?view). 

I spent around 6-7 hours to fix some. Here is the result: https://hackmd.io/KYNgrAjGDMAmDGBaUxiICwCYAMtEA5tp9EBDdAMxwCMBOCbMeCoA?view

Here is the diff:
https://www.diffchecker.com/Nl5cxDew

You may notice that some new issues are added, but do not worry, many of them are false-positives and there are no other way to fix the issue.

It is the first PR in series of PRs, which fix all problems.

Here I fixed only:
- iroha-cli
- libs
- ametsuchi
- consensus

### Benefits

This PR does not change any logic, it introduces only small code refactorings.

Generally cppclean looks for unused includes, or suggests to place forward declarations. Both of these techniques reduce interdependency between objects, which is good -- it reduces recompilation of targets, if certain headers are changed.

Also, it found functions defined in cpp without keyword `static`. Without this keyword, functions have `external linkage` by default and are accessible by the rest of program just through forward declaration. I fixed also this.

### Possible Drawbacks 

There are many issues with logger. Because logger is an alias to `spdlog::logger`, I need to write many lines to write forward declaration, which is bad. 
If we really need to fix these issues, I suggest to remove `logger::Logger` and use `std::unique_ptr<stdlog::logger>` everywhere. For now logger introduces many false positives.

### CODE REVIEW

**Comment by Warchant** [comment]
storage.hpp before:
image

storage.hpp after:
image

**Comment by Warchant** [comment]
storage_result.hpp before:
image

storage_result.hpp after:
image

File: irohad/ametsuchi/impl/mutable_storage_impl.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include "ametsuchi/impl/postgres_wsv_query.hpp"
 #include "ametsuchi/impl/redis_block_index.hpp"
 #include "model/commands/transfer_asset.hpp"
-
+#include "model/execution/command_executor_factory.hpp"  // for iroha::model::CommandExecutorFactory
```
Comment (Alexey-N-Chernyshov) [inline irohad/ametsuchi/impl/mutable_storage_impl.cpp]: Do we need this comment?

File: irohad/ametsuchi/impl/peer_query_wsv.cpp [hunk 2]
```diff
@@ -15,9 +15,10 @@
  * limitations under the License.
  */
 
-#include <utility>
-
 #include "ametsuchi/impl/peer_query_wsv.hpp"
+#include <utility>
```
Comment (Alexey-N-Chernyshov) [inline irohad/ametsuchi/impl/peer_query_wsv.cpp]: I'd prefer to order includes like Google (https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes).

File: irohad/ametsuchi/impl/postgres_wsv_command.hpp [hunk 3]
```diff
@@ -18,11 +18,24 @@
 #ifndef IROHA_POSTGRES_WSV_COMMAND_HPP
 #define IROHA_POSTGRES_WSV_COMMAND_HPP
 
-#include "ametsuchi/wsv_command.hpp"
-#include "postgres_wsv_common.hpp"
+#include <set>
+#include <string>
+#include "ametsuchi/impl/postgres_wsv_common.hpp"
+#include "ametsuchi/wsv_command.hpp" // for WsvCommand
```
Comment (Alexey-N-Chernyshov) [inline irohad/ametsuchi/impl/postgres_wsv_command.hpp]: Do we need a comment?

File: irohad/main/impl/consensus_init.cpp [hunk 4]
```diff
@@ -16,12 +16,12 @@
  */
 
 #include "main/impl/consensus_init.hpp"
-
```
Comment (Alexey-N-Chernyshov) [inline irohad/main/impl/consensus_init.cpp]: I'd keep this line to separate related header like (https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes).

File: irohad/model/commands/all.hpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
```
Comment (Alexey-N-Chernyshov) [inline irohad/model/commands/all.hpp]: 2018

File: irohad/ametsuchi/impl/mutable_storage_impl.cpp (hunk 1 above)
Comment (Warchant) [reply irohad/ametsuchi/impl/mutable_storage_impl.cpp]: Comments help me to not forget why exactly this header is included. It is just a coincidence that header name and symbol name are equal. In many cases they are not.

File: irohad/ametsuchi/impl/postgres_wsv_command.hpp (hunk 3 above)
Comment (Warchant) [reply irohad/ametsuchi/impl/postgres_wsv_command.hpp]: I answered above.

File: irohad/consensus/yac/storage/yac_block_storage.hpp [hunk 6]
```diff
@@ -62,12 +63,12 @@ namespace iroha {
         /**
          * @return votes attached to storage
          */
-        auto getVotes() -> decltype(votes_);
+        std::vector<VoteMessage> getVotes();
```
Comment (nickaleks) [inline irohad/consensus/yac/storage/yac_block_storage.hpp]: Why is next method declared const and this one is not?

File: irohad/consensus/yac/cluster_order.hpp [hunk 7]
```diff
@@ -54,20 +55,17 @@ namespace iroha {
          */
         bool hasNext();
 
-        std::vector<model::Peer> getPeers() {
-          return order_;
-        };
+        std::vector<model::Peer> getPeers();
```
Comment (nickaleks) [inline irohad/consensus/yac/cluster_order.hpp]: This and the next method should be const?

File: irohad/consensus/yac/storage/impl/yac_block_storage.cpp [hunk 8]
```diff
@@ -56,11 +54,11 @@ namespace iroha {
         return getState();
       }
 
-      auto YacBlockStorage::getVotes() -> decltype(votes_) {
+      std::vector<VoteMessage> YacBlockStorage::getVotes() {
```
Comment (nickaleks) [inline irohad/consensus/yac/storage/impl/yac_block_storage.cpp]: This should be const as well

File: irohad/model/commands/add_asset_quantity.hpp [hunk 9]
```diff
@@ -33,18 +32,18 @@ namespace iroha {
       /**
        * Account where to add assets
        */
-      std::string account_id{};
+      std::string account_id;
```
Comment (nickaleks) [inline irohad/model/commands/add_asset_quantity.hpp]: What is the reason behind removal of initialization?

Comment (Warchant) [reply irohad/model/commands/add_asset_quantity.hpp]: I see only one advantage of using that initialization -- it saves you some code and time when you have to initialize many vars in many constructors.

However, one disadvantage is that it decreases overall coverage -- std::string may not be constructed and this line is marked as partial.

**Pull request merged**